src/blackbox/reader.go

```diff
@@ -64,23 +61,27 @@ func (f *FlightLogReader) LoadFile(file io.Reader, ctx context.Context) (<-chan
 			default:
 			}
 
-			frame, err := frameReader.ReadNextFrame()
-			if err != nil {
-				// the frame is corrupted
-				errChan <- err
-				// we should skip all bytes until the next frame
+			frame := frameReader.ReadNextFrame()
+			// if the frame is corrupted - we should skip all bytes until the next frame
+			if frame.Error() != nil {
 				_, err = consumeToNext(decoder) // @TODO: consume skippedFrames value if needed
-				if err != nil {
-					errChan <- err
-				}
-				continue
+			}
+			if err != nil {
```

Comment:
What do you think of
```
if frame.Error() != nil {
  // Try to consume the remaining bytes from the broken frame
  if err := consumeToNext(decoder); err!=nil{
    frame.setError(err)
    frameChan <- frame
    return
  }
}
```
?

That way we don't need to understand what's done with the method-scoped `err` variable.

Reply:
btw, question - do we plan to consume that int64 value, returned from `consumeToNext`?
https://github.com/maxlaverse/blackbox-library/blob/master/src/blackbox/reader.go#L106
it was there before, and I didn't remove it, but we actually never used it

Reply:
We should track the number of bytes that were dropped but where and how I'm not sure yet.

Most of statistics are generated in `FrameReader.PreComplete()`. We could move `consumeToNext()` in `FrameReader` to have them all there, and update a counter with the number of dropped bytes.

Another solution would be to generate the statistics in `FlightLogReader` instead, and move the statistics there.

Or something in the middle, move `consumeToNext()` to `FrameReader` but have it return a Frame with the remaining bytes, and the right `Start` and `Stop` offset for `FlightLogReader` to update its statistics